spark-statsd-reporter/src/main/java/org/apache/spark/metrics/sink/statsd/StatsdReporter.java

```diff
@@ -112,66 +88,71 @@ private void reportCounters(SortedMap<String, Counter> counters, DatagramSocket
+        	addIfMatches(metrics, MetricAttribute.M1_RATE, metricFormatter.buildMetricString(name, MetricAttribute.M1_RATE.toString(), convertRate(meter.getOneMinuteRate()), TIMER));
+        	addIfMatches(metrics, MetricAttribute.M5_RATE, metricFormatter.buildMetricString(name, MetricAttribute.M5_RATE.toString(), convertRate(meter.getFiveMinuteRate()), TIMER));
+        	addIfMatches(metrics, MetricAttribute.M15_RATE, metricFormatter.buildMetricString(name, MetricAttribute.M15_RATE.toString(), convertRate(meter.getFifteenMinuteRate()), TIMER));
+        	addIfMatches(metrics, MetricAttribute.MEAN_RATE, metricFormatter.buildMetricString(name, MetricAttribute.MEAN_RATE.toString(), convertRate(meter.getMeanRate()), TIMER));
+            send(socket, metrics.toArray(new String[metrics.size()]));
         });
     }
 
     private void reportTimers(SortedMap<String, Timer> timers, DatagramSocket socket) {
         timers.forEach((name, timer) -> {
             Snapshot snapshot = timer.getSnapshot();
-            send(socket,
-                    metricFormatter.buildMetricString(name, "max", convertDuration(snapshot.getMax()), TIMER),
-                    metricFormatter.buildMetricString(name, "mean", convertDuration(snapshot.getMean()), TIMER),
-                    metricFormatter.buildMetricString(name, "min", convertDuration(snapshot.getMin()), TIMER),
-                    metricFormatter.buildMetricString(name, "stddev", convertDuration(snapshot.getStdDev()), TIMER),
-                    metricFormatter.buildMetricString(name, "p50", convertDuration(snapshot.getMedian()), TIMER),
-                    metricFormatter.buildMetricString(name, "p75", convertDuration(snapshot.get75thPercentile()), TIMER),
-                    metricFormatter.buildMetricString(name, "p95", convertDuration(snapshot.get95thPercentile()), TIMER),
-                    metricFormatter.buildMetricString(name, "p98", convertDuration(snapshot.get98thPercentile()), TIMER),
-                    metricFormatter.buildMetricString(name, "p99", convertDuration(snapshot.get99thPercentile()), TIMER),
-                    metricFormatter.buildMetricString(name, "p999", convertDuration(snapshot.get999thPercentile()), TIMER),
-                    metricFormatter.buildMetricString(name, "m1_rate", convertRate(timer.getOneMinuteRate()), TIMER),
-                    metricFormatter.buildMetricString(name, "m5_rate", convertRate(timer.getFiveMinuteRate()), TIMER),
-                    metricFormatter.buildMetricString(name, "m15_rate", convertRate(timer.getFifteenMinuteRate()), TIMER),
-                    metricFormatter.buildMetricString(name, "mean_rate", convertRate(timer.getMeanRate()), TIMER)
-            );
+            List<String> metrics = new ArrayList<>();
+            addIfMatches(metrics, MetricAttribute.MAX, metricFormatter.buildMetricString(name, MetricAttribute.MAX.toString(), convertDuration(snapshot.getMax()), TIMER));
```

Comment:
`MetricAttribute.MAX.toString()` will return stringified enum value, `"MAX"` in this case. `getCode` should be used instead of `toString` for `MetricAttribute` values.